frontend/src/components/AccountDetailsEditContainer.vue

```diff
@@ -0,0 +1,259 @@
+                  Adminaccount
+                </label>
+              </div>
+            </td>
+          </tr>
+        </table>
+        <div class="flex items-center pt-4">
+          <ButtonComponent
+            class="w-fit p-1"
+            :loading="disableInputShowSpinner"
+            @click="reset()"
+          >
+            Abbrechen
+          </ButtonComponent>
+          <div class="flex-grow text-center">
+            <div class="text-red-500 italic" v-if="displayError">
+              Ungültige Eingabe
+            </div>
+          </div>
+          <ButtonComponent
+            class="w-fit p-1"
+            :loading="disableInputShowSpinner"
+            @click="updateInformation()"
+          >
+            Speichern
+          </ButtonComponent>
+        </div>
+        <div :class="showadminview ? 'flex items-center pt-1' : 'collapse'">
+          <div class="flex-grow text-center">
+            <div class="text-green-500 italic" v-if="deleted">
+              Account gelöscht
+            </div>
+          </div>
+          <ButtonComponent
+            class="w-fit p-1"
+            :loading="disableInputShowSpinner"
+            @click="deleteAccount()"
+          >
+            Löschen
+          </ButtonComponent>
+        </div>
+      </template>
+    </ContainerComponent>
+  </div>
+</template>
+<script setup lang="ts">
+import { ref } from 'vue';
+import TextInputComponent from './TextInputComponent.vue';
+import ButtonComponent from './ButtonComponent.vue';
+import ContainerComponent from './ContainerComponent.vue';
+import { IUser, UserClass } from '../../../model/user/IUser';
+import { AccountRestInterface } from '../restInterfaces/AccountRestInterface';
+import {
+  LoernwerkError,
+  LoernwerkErrorCodes,
+} from '../../../backend/loernwerkError';
+import { router } from '../router';
+
+const props = defineProps({
+  /**
+   * Whether the delete button is displayed
+   */
+  showadminview: {
+    type: Boolean,
+    required: false,
+    default: false,
+  },
+  userid: {
+    type: Number,
+    required: false,
+    default: null,
+  },
+});
+
+const nameField = ref('');
+const mailField = ref('');
+const pwField = ref('');
+const pwFieldControl = ref('');
+const disableInputShowSpinner = ref(false);
+const deleted = ref(false);
+const displayError = ref(false);
+const forbidden = ref(false);
+let originalUser: Partial<IUser>;
+try {
+  originalUser =
+    props.userid === null
+      ? await AccountRestInterface.getOwnAccount()
+      : await AccountRestInterface.getAccount(props.userid);
+} catch (e) {
+  if (e instanceof LoernwerkError) {
+    if (e.code === LoernwerkErrorCodes.UNAUTHORIZED) {
+      router.push('LogIn');
+    } else if (e.code === LoernwerkErrorCodes.FORBIDDEN) {
+      forbidden.value = true;
+    } else {
+      throw e;
+    }
+  } else {
+    throw e;
+  }
+} finally {
+  originalUser = { type: UserClass.REGULAR };
```

Comment:
What is this for specifically?

Reply:
hmm This does not make any sense and will be removed soon